diff -ru a/modules/dns/dns.tcl b/modules/dns/dns.tcl
--- a/modules/dns/dns.tcl	2018-01-26 05:36:55.000000000 +0100
+++ b/modules/dns/dns.tcl	2019-12-02 17:17:18.062108231 +0100
@@ -227,6 +227,10 @@
     set state(-cafile)     $options(cafile);    # certificate authority file
     set state(-cadir)      $options(cadir);     # certificate authority dir
 
+    if {[info exists state(mdns)]} {
+        unset state(mdns)
+    }
+
     # Handle DNS URL's
     if {[string match "dns:*" $query]} {
         array set URI [uri::split $query]
@@ -284,6 +288,14 @@
             return -code error "udp support is not available,\
                 get ceptcl or tcludp"
         }
+
+        # RFC 6762 mDNS multicast addresses
+        if {$state(-port) == 5353} {
+            if {$state(-nameserver) eq "224.0.0.251" \
+                || [string tolower $state(-nameserver)] eq "\[ff02::fb\]"} {
+                set state(mdns) {}
+            }
+        }
     }
 
     # Check for reverse lookups
@@ -817,21 +829,28 @@
     global errorInfo errorCode
 
     if {[string length $errormsg] != 0} {
-	set state(error) $errormsg
-	set state(status) error
+        set state(error) $errormsg
+        set state(status) "error"
     }
-    catch {close $state(sock)}
-    catch {after cancel $state(after)}
+
     if {[info exists state(-command)] && $state(-command) != {}} {
-	if {[catch {eval $state(-command) {$token}} err]} {
-	    if {[string length $errormsg] == 0} {
-		set state(error) [list $err $errorInfo $errorCode]
-		set state(status) error
-	    }
-	}
+        if {[catch {eval $state(-command) {$token}} err]} {
+            if {[string length $errormsg] == 0} {
+                set state(error) [list $err $errorInfo $errorCode]
+                set state(status) "error"
+            }
+        }
+    }
+
+    if {![info exists state(mdns)] || $state(status) eq "error"} {
+        catch {close $state(sock)}
+        catch {after cancel $state(after)}
+
         if {[info exists state(-command)]} {
             unset state(-command)
         }
+    } elseif {[info exists state(reply)]} {
+        unset state(reply)
     }
 }
 
@@ -952,6 +971,7 @@
     set id [expr {$id & 0xFFFF}]
     if {$id != [expr {$state(id) & 0xFFFF}]} {
         ${log}::error "received packed with incorrect id"
+        return
     }
     # bug #1158037 - doing this causes problems > 65535 requests!
     #Receive [namespace current]::$id
